web/annotation_layer_builder.js

```diff
@@ -33,6 +33,7 @@ import { SimpleLinkService } from "./pdf_link_service.js";
  * @property {IL10n} l10n - Localization service.
  * @property {boolean} [enableScripting]
  * @property {Promise<boolean>} [hasJSActionsPromise]
+ * @property {Promise<Array<Object> | null} [fieldObjectsPromise]
```

Comment:
The type is missing a `>` after `null` (`Promise<` is not closed). Not just here, but everywhere else in this patch.

`fieldObjects` is an object, not an array of objects. A type of `object` is rather unclear. Is it possible to improve the documentation of the public API, so that it is more obvious what its value should be? Even if it just states something like "The document's fieldObjects" at [
https://github.com/mozilla/pdf.js/blob/680f33c31c23c90d01065695e9b278748ad8ccfc/src/core/document.js#L1407-L1435](https://github.com/mozilla/pdf.js/blob/680f33c31c23c90d01065695e9b278748ad8ccfc/src/core/document.js#L1407-L1435)

Reply:
> The type is missing a `>` after `null` (`Promise<` is not closed). Not just here, but everywhere else in this patch.

Good eye, that was a copy-and-paste error on my part.

*Regarding the pre-existing type issue:*
I've tried to improve the type in the API to outline the data format a bit better, and copied that across the patch, however I'm not convinced that also linking to the worker-thread code helps all that much (you'd probably need to look at the various `getFieldObject`-methods in `src/core/annotation.js` too).

---

Thanks for the review!